app/config.go

```diff
@@ -40,3 +40,8 @@ func GetDelayedStopInterval() string {
 	}
 	return ""
 }
+
+type LifecycleAware interface {
+	OnStartup()
```

Comment:
I am thinking onStartup and onShutdown may duplicate with start and stop.  Can we just have LifecycleAware interface with a method(LifecycleAwared) to indicate it is LifecycleAware trigger and use start and stop method like the normal trigger.

Reply:
This is not a management interface... Trigger already has a start/stop.  This is more of a listener interface, so we should keep them separate.

Reply:
Oh, yes.  Those are additional interface out of management and co-exist with start and stop